core/src/avm2.rs

```diff
@@ -609,6 +614,53 @@ impl<'gc> Avm2<'gc> {
             panic!("SymbolClass ClassObject and Class don't match");
         }
 
+        let library = activation.context.library.library_for_movie_mut(movie);
+
+        let character = library.character_by_id(id);
+
+        if let Some(character) = character {
+            if matches!(
+                character,
+                Character::EditText(_)
+                    | Character::Graphic(_)
+                    | Character::MovieClip(_)
+                    | Character::Avm2Button(_)
+            ) {
+                // The class must extend DisplayObject to ensure that events
+                // can properly be dispatched to them
+                if !class.has_class_in_chain(
+                    activation
+                        .avm2()
+                        .classes()
+                        .display_object
+                        .inner_class_definition(),
+                ) {
+                    return Err(Error::AvmError(type_error(
+                        activation,
+                        &format!("Error #2022: Class {}$ must inherit from DisplayObject to link to a symbol.", name.to_qualified_name(activation.gc())),
+                        2022,
+                    )?));
+                }
+            }
+        } else if movie_clip.avm2_class().is_none() {
+            // If this ID doesn't correspond to any character, and the MC that
```

Comment:
What happens if multiple entries don't correspond to characters? (like 0, 12345, 23456)

Reply:
This was already resolved in #17275 - FP silently ignores them